config/confighttp/confighttp.go

```diff
@@ -119,6 +120,21 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 	}, nil
 }
 
+func (hcs *HTTPClientSettings) WithDefaultUserAgent(defaultUserAgent string) {
```

Comment:
Please comment on how to use this, also a bit strange that is not an option in the "ToClient" but a func that changes the internal state.

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/pull/3970#discussion_r702134406

Adding it to the ToClient interface was my initial thought as well, though I also see why changing that interface might be undesirable.

Reply:
I think we should start adding "Options" to that func and this can be one of the first option we create.

Also, this can be done by the caller already since they have access to the "Headers", how useful is this? How many callers do we have? 

I am trying to limit the API unless we have a clear reason to have it.

Reply:
Adding options to `ToClient` would also be a way to handle this.  I initially proposed a separate method to avoid changing the existing interface at all, but adding a variadic parameter is as close to no change as you can get since it should not affect existing callers.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/pull/3970#issuecomment-936850553